lua/wire/clientsidehttp.lua

```diff
@@ -0,0 +1,382 @@
+    		for i,entry in ipairs(clHTTP.requestWhitelist.clientside[client]) do
+    			if url:match(entry) then
+    				return true,"clientside"
+    			end
+    		end
+    	end
+
+    	for i,entry in ipairs(clHTTP.requestWhitelist.serverside) do
+    		if url:match(entry) then
+    			return true,"serverside"
+    		end
+    	end
+
+    	return false
+    end
+
+    -- void function(entity client, string url, function callback_success[, function callback_failure])
+    function clHTTP.request(client,url,callback_success,callback_failure)
+    	-- When the client sends us the data from the request, we need to know which
+    	-- request it was for, so a UID is used.
+
+    	if not clHTTP.canRequest(client,url) then return end
+
+    	if client and (cvar_useClient:GetInt() ~= 0) then
+    		local uid
+    		repeat
+    			uid = newUID()
+    		until canOverwriteUID(uid)
+
+    		clHTTP.requests[uid] = {
+    			success = callback_success,
+    			failure = callback_failure,
+    			lastTouched = SysTime(),
+    		}
+
+    		launchNewRequest(client,url,uid)
+    	else
+    		clHTTP.rawRequest(url,callback_success,callback_failure)
+    	end
+    end
+
+    -- The serverside whitelist reloading function
+    -- void function(entity ply)
+    local function reloadWhitelist(ply)
+    	if IsValid(ply) then return end
+
+    	local whitelistText = file.Read("wire_http_whitelist_sv.txt","DATA")
+    	if whitelistText and (whitelistText ~= "") then
+    		for entry,comment in whitelistText:gmatch("([^#\n\r]*)[^\n\r]*[\n\r]*") do
+    			if entry ~= "" then
+    				clHTTP.requestWhitelist.serverside[#clHTTP.requestWhitelist.serverside+1] = entry
+    			end
+    		end
+    	else
+    		file.Write("wire_http_whitelist_sv.txt",clHTTP.defaultWhitelistData)
+    	end
+
+    	print("Server-side HTTP whitelist reloaded")
+    end
+    concommand.Add("wire_expression2_http_reloadwhitelist",reloadWhitelist)
+
+    reloadWhitelist()
+
+    net.Receive(clHTTP.netMsgID,function(_,ply)
+    	handleIncomingRequest(ply)
+    end)
+
+    net.Receive(clHTTP.whitelistSyncNetMsgID,function(_,ply)
+    	local whitelistData = net.ReadTable()
+
+    	local whitelist = {} -- create (and reset) the table
+
+    	for i,entry in ipairs(clHTTP.requestWhitelist.clientside.default) do
+    		whitelist[#whitelist+1] = entry
+    	end
+
+    	for i,entry in ipairs(whitelistData) do
+    		whitelist[#whitelist+1] = entry
+    	end
+
+    	clHTTP.requestWhitelist.clientside[ply] = whitelist
+    end)
+else
+    clHTTP.defaultWhitelistData = [[#
+    #	The Wiremod Expression 2 client-side HTTP request whitelist
+    #
+    #	Hints:
+    #	- One entry per line!
+    #	- Entries are also lua string patterns
+    #	- Anything after a # in an entry is ignored and treated as a comment
+    #	- You do not have to have a new line at the end of this file.
+    #	- Empty lines are ignored
+    #
+    #	Have fun!
+
+    #^https?://www.github.com/.*$
+    #^https?://www.github.com/.*$
+    ]]
```

Comment:
Is the difference really just the word "client"/"server"?
Can't you put placeholders in there and put it at the top for both?
````Lua
local defaultWhitelistData = [[...
...
````
clHTTP.defaultWhitelistData  = defaultWhitelistData.gsub('CLIENT_OR_SERVER_PLACEHOLDER', 'client')